walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,107 @@
+    if(request.getDuration() > (24*60)) {
+      return conflicts;
+    }
+
+    // Find conflicts for all the required attendees.
+    for(Event event : events) {
+      for(String attendee : request.getAttendees()){
+        if(event.getAttendees().contains(attendee)) {
+          conflicts.add(event.getWhen());
+        }
+      }
+    }
+
+    // If there are no conflicts, the whole day is open
+    if(conflicts.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // Sort the conflicts
+    Collections.sort(conflicts, TimeRange.ORDER_BY_START);
+
+    // Combine the conflicts if there's no room for breaks between them   
+    ArrayList<TimeRange> combinedConflicts = new ArrayList<>(); 
+    if(conflicts.size() == 1) {
+      combinedConflicts.add(conflicts.get(0));
+    } else {
+      for(int i = 1; i < conflicts.size(); i++) {
+        TimeRange currentTimeRange = (TimeRange)conflicts.get(i);
+        TimeRange previousTimeRange = (TimeRange)conflicts.get(i-1);
+
+        // Let's check & see if two conflicts overlap with each other
+        if (currentTimeRange.overlaps(previousTimeRange)) {
+          // If the two events are nested, get the larger event, else, combine the conflicts
+          if(currentTimeRange.contains(previousTimeRange) || previousTimeRange.contains(currentTimeRange)) {
+            if(currentTimeRange.contains(previousTimeRange)) {
+              combinedConflicts.add(currentTimeRange);
+            } else if (previousTimeRange.contains(currentTimeRange)) {
+              combinedConflicts.add(previousTimeRange);
+            }
+          } else {
+            TimeRange combinedTimeRange = TimeRange.fromStartEnd(
+              previousTimeRange.start(), currentTimeRange.end(), false);
+            combinedConflicts.add(combinedTimeRange);
+          }
+          continue;
+        }
+        combinedConflicts.add(currentTimeRange);
+        // On our first run through let's add the previousTimeRange if it isn't overlapped or nested
+        if(i==1) {
+          combinedConflicts.add(previousTimeRange);
+        }
+      }
+    }
+
+    // Sort the collection one more time
+    Collections.sort(combinedConflicts, TimeRange.ORDER_BY_START);
+
+    // If the first conflict isn't at START_OF_DAY, we can add our first timeslot
+    if(combinedConflicts.get(0).start() != TimeRange.START_OF_DAY) {
+      meetingTimes.add(TimeRange.fromStartEnd(
+        TimeRange.START_OF_DAY, combinedConflicts.get(0).start(), false));
+    }
+    // Remove conflict timeslots from meetingTimes
+    for(int i = 1; i < combinedConflicts.size(); i++){
+      meetingTimes.add(TimeRange.fromStartEnd(
+        combinedConflicts.get(i-1).end(), combinedConflicts.get(i).start(), false));
+    }
+    // If the last conflict doesn't end at END_OF_DAY, we can add the final timeslot
+    if(combinedConflicts.get(combinedConflicts.size()-1).end() != TimeRange.END_OF_DAY) {
```

Comment:
nit: generally you want spaces in around operator usage, so  combinedConflicts.size() - 1.
Same comment for other parts of this code.